relay-ourlogs/src/vercel_to_sentry.rs

```diff
@@ -0,0 +1,722 @@
+//! Transforms the Vercel Log Drain format to Sentry Logs.
+
+use std::str::FromStr;
+
+use chrono::{TimeZone, Utc};
+use relay_conventions::{ENVIRONMENT, HTTP_RESPONSE_STATUS_CODE, ORIGIN, SERVER_ADDRESS, URL_PATH};
+use relay_event_schema::protocol::{Attributes, OurLog, OurLogLevel, SpanId, Timestamp, TraceId};
+use relay_protocol::{Annotated, Meta, Remark, RemarkType};
+use serde::{Deserialize, Serialize};
+
+/// Log Level for Vercel Logs
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum VercelLogLevel {
+    /// Info level logs.
+    Info,
+    /// Warning level logs.
+    Warning,
+    /// Error level logs.
+    Error,
+    /// Fatal level logs.
+    Fatal,
+}
+
+/// Vercel proxy information for requests.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VercelProxy {
+    /// Unix timestamp when the proxy request was made.
+    pub timestamp: i64,
+    /// HTTP method of the request.
+    pub method: String,
+    /// Hostname of the request.
+    pub host: String,
+    /// Request path with query parameters.
+    pub path: String,
+    /// User agent strings of the request.
+    #[serde(rename = "userAgent")]
```

Comment:
You can put `#[serde(rename_all = "camelCase")]` on the struct to prevent having a `rename` on every field.